misk-aws/src/main/kotlin/misk/jobqueue/sqs/SqsJobConsumer.kt

```diff
@@ -63,73 +71,120 @@ internal class SqsJobConsumer @Inject internal constructor(
   }
 
   internal inner class QueueReceiver(
-    private val queueName: QueueName,
+    queueName: QueueName,
     private val handler: JobHandler
   ) {
     private val queue = queues[queueName]
 
-    fun runOnce(): Status {
-      val (receiveDuration, messages) = queue.call { client ->
-        val receiveRequest = ReceiveMessageRequest()
-          .withAttributeNames("All")
-          .withMessageAttributeNames("All")
-          .withQueueUrl(queue.url)
-          .withMaxNumberOfMessages(config.message_batch_size)
+    fun run(): Status {
+      // Receive messages in parallel. Default to 1 if feature flag is not defined.
+      val numReceivers = receiversForQueue()
+      val futures = (1..numReceivers)
+          .filter { num ->
+            val lease = leaseManager.requestLease("sqs-job-consumer-${queue.name.value}-$num")
+            !config.clustered_consumers || lease.checkHeld()
+          }
+          .map {
+            CompletableFuture.supplyAsync(Supplier {
+              receive()
+            }, receivingThreads)
+          }
 
-        timed { client.receiveMessage(receiveRequest).messages }
+      // Either all messages are consumed and processed successfully, or we signal failure.
+      // If none of the received consume any messages, return NO_WORK for backoff.
+      return CompletableFuture.allOf(*futures.toTypedArray()).thenApply {
+        futures.flatMap { it.join() }.minBy {
+          when (it) {
+            Status.FAILED -> 1
```

Comment:
this is weird. why are we converting a status enum to some arbitrary number?

Reply:
if at least one consumer failed I want to return`Status.FAILED`. If there was at least one message consume return OK, if nobody worked I will return `NO_WORK`.  So I map every status to a a number I'm trying to find the `min`. Happy to change if you think there's a simpler way.

Reply:
I would just build a custom reducer that merges status results